elections/uk/management/commands/uk_create_elections_from_every_election.py

```diff
@@ -7,267 +7,36 @@
-                name=area_type.name).exists():
-            election.area_types.add(area_type)
-
-        return area_type
-
-    def add_area(self, area_info, election, organization_extra):
-        """
-        Case 1: An 'area' is the area (like a council ward) that has a ballot
-        paper, not necessarily the area of the organisation with an election.
-
-        DC Elections will attach a 'division' to the ballot paper ID that
-        relates to the division that was active on the date of the election
-
-        Case 2: When there is a single ballot paper for the organisation, the
-        area is that of the the organisation. For example, for a mayoral or PCC
-        election.
-        """
-
-        party_set = self.get_party_set(area_info)
-
-        area_type = self.get_area_type(area_info, election)
+class Command(BaseCommand):
+    help = 'Create posts and elections from a EveryElection'
 
-        if area_info['division']:
-            # Case 1, there is an organisational division relted to this
-            # area/post
-            area_name = area_info['division']['name']
-            post_role = area_info['division']['geography_curie']
-            post_extra_slug = ":".join([
-                area_info['division']['division_type'],
-                area_info['division']['geography_curie'].split(':')[-1]
-            ])
-            area_identifier = area_info['division']['geography_curie']
-            winner_count = area_info['division']['seats_total']
-        else:
-            # Case 2, this organisation isn't split in to divisions for this
-            # election, take the info from the organisation directly
-            area_name = area_info['organisation']['official_name']
-            post_role = area_info['elected_role']
-            post_extra_slug = area_info['organisation']['slug']
-            area_identifier = ":".join([
-                area_info['organisation']['organisation_type'],
-                area_info['organisation']['slug'],
-            ])
-            # TODO this might not always be true, get the count from EE
-            winner_count = 1
 
-        try:
-            post_extra = PostExtra.objects.get(slug=post_extra_slug)
-            post = post_extra.base
-        except PostExtra.DoesNotExist:
-            post = Post.objects.create(
-                label=area_name,
-                organization=organization_extra.base,
-            )
-            post_extra = PostExtra.objects.create(
-                base=post,
-                slug=post_extra_slug,
-            )
+    def handle(self, *args, **options):
 
-        post.role = post_role
-        post.label = area_name
-        post.organization = organization_extra.base
-        post.save()
-        post_extra.party_set = party_set
-        post_extra.save()
+        # Get all elections from EveryElection
+        ee_importer = EveryElectionImporter()
+        ee_importer.build_election_tree()
+
+        with transaction.atomic():
```

Comment:
Nice. I expect that by doing all the API calls to build the data structure first and then doing all the database IO in a transaction afterwards, that will probably fix (or massively alleviate) https://github.com/DemocracyClub/yournextrepresentative/issues/302